src/main/java/com/example/springresttask/repository/EmailRepositoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.example.springresttask.repository;
+
+import com.example.springresttask.domain.Email;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class EmailRepositoryImpl implements EmailRepository {
+
+    private final EntityManager entityManager;
+
+    @Transactional
+    @Override
+    public List<Email> findAllByPendingEmail() {
+        TypedQuery<Email> q = entityManager.createQuery("""
+                        SELECT e
+                        FROM email e
+                        WHERE e.isSent = false
+                        """,
+                Email.class);
+        return q.getResultList();
+    }
+
+    @Override
+    public void deletePendingEmail(Long id) {
```

Comment:
When I put a comment about transaction it means that in other places situation is the same :)